### PR TITLE
refactor(j2cl): split compose test doubles + rich-component tests out of monster class (#1270)

### PR DESCRIPTION
**Pull request by vega113**
Closes #1270 (compose split).

`J2clComposeSurfaceControllerTest` was the ~5 kLOC monster class #1270 calls out. This PR fully splits it — pure, behavior-preserving, no production code changes.

## What changed
- **New `org.waveprotocol.box.j2cl.testsupport` package** — the stateless doubles (`FakeGateway`, `FakeView`, `FakeAttachmentTransport`, `CapturingDeltaFactory`, `FakeFactory`) moved here as **public** classes (no `@J2clTestInput`; only the owning `*Test` entry points carry it).
- **`J2clComposeControllerTestSupport`** (abstract, package-private) — shared controller builders + common assertions, so the split classes don't duplicate scaffolding.
- **15 focused per-flow test classes**, each `extends J2clComposeControllerTestSupport` and each **< 800 LOC**:

  | Class | tests | LOC |
  |---|--:|--:|
  | J2clComposeStaleGenerationRetryTest | 22 | 512 |
  | J2clComposeRichToolbarFormattingTest | 19 | 463 |
  | J2clComposeAttachmentGuardsTest | 18 | 451 |
  | J2clComposeAttachmentCancelLifecycleTest | 16 | 524 |
  | J2clComposeCreateFlowTest | 14 | 362 |
  | J2clComposeReplyEditFlowTest | 13 | 346 |
  | J2clComposeAttachmentUploadTest | 13 | 498 |
  | J2clComposeTaskReactionTest | 13 | 357 |
  | J2clComposeSelectedWaveContextTest | 10 | 231 |
  | J2clComposeDropAndDeleteBlipTest | 9 | 302 |
  | J2clComposeMentionsTest | 8 | 239 |
  | J2clComposeWaveHeaderActionsTest | 8 | 262 |
  | J2clComposeRichReplyFailureRetryTest | 7 | 224 |
  | J2clComposeRichComponentTest | 6 | 278 |
  | J2clComposeSignedOutLifecycleTest | 6 | 151 |

- Deleted the original monster file and the interim `ComposeTestDoubles.java`.
- Documented the new `testsupport` package in `docs/j2cl-gwttestcase-verification-matrix.md`.

## Verification
- `mvnw -f j2cl/pom.xml test-compile` → **BUILD SUCCESS** (74 test sources).
- `surefire:test -Dtest='J2clCompose*Test'` → **182 tests, 0 failures** across the 15 classes — identical coverage to the pre-split file.

## Acceptance criteria (#1270)
- ✅ All existing test methods still pass (no behavior change) — 182/182.
- ✅ Each new split file < 800 LOC (max 524).
- ✅ Shared fakes moved to a `testsupport` package; reflection-based `CapturingDeltaFactory` preserved.
- ✅ A developer can add a focused compose edge-case test without touching a 5 kLOC file.

**Follow-up:** the sibling `J2clSelectedWaveProjectorTest` (~4.1 kLOC) split named in #1270 is tracked separately to keep this PR reviewable (single-lane, as the issue requests).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: j2cl/src/test/java/org/waveprotocol/box/j2cl/compose/ComposeTestDoubles.java [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+import org.waveprotocol.box.j2cl.transport.SidecarSubmitRequest;
+import org.waveprotocol.box.j2cl.transport.SidecarSubmitResponse;
+
+// #1270: the shared compose test doubles, moved out of the 5 kLOC
+// J2clComposeSurfaceControllerTest so they can be reused by focused per-flow
+// split test classes. Kept in this package (not testsupport) so the existing
+// package-private field access from the test classes is preserved verbatim —
+// these are stateless doubles that need no @J2clTestInput.
+
+final class FakeGateway implements J2clComposeSurfaceController.Gateway {
```
[the remark this answers is not in the document]
Comment (vega113) [reply j2cl/src/test/java/org/waveprotocol/box/j2cl/compose/ComposeTestDoubles.java]: Fixed in 995a7f4 — the doubles now live as public classes in a new `org.waveprotocol.box.j2cl.testsupport` package (FakeGateway/FakeView/FakeAttachmentTransport/CapturingDeltaFactory/FakeFactory), carrying no `@J2clTestInput`. The interim ComposeTestDoubles.java is deleted.

File: j2cl/src/test/java/org/waveprotocol/box/j2cl/compose/J2clComposeSurfaceControllerTest.java [hunk 2]
```diff
@@ -4771,490 +4771,5 @@ private static void assertContains(String value, String... expectedSubstrings) {
...
-    // The space run is emitted as plain text — no annotation start
-    // wrapping a whitespace-only payload (which the builder would
-    // reject as "Missing annotated text.").
-    String delta = gateway.lastSubmitRequest.getDeltaJson();
-    assertContains(delta, "\"2\":\"hi\"");
-    assertContains(delta, "\"2\":\" \"");
-    assertContains(delta, "\"2\":\"there\"");
-  }
-
-  private static final class FakeGateway implements J2clComposeSurfaceController.Gateway {
```
[the remark this answers is not in the document]
Comment (vega113) [reply j2cl/src/test/java/org/waveprotocol/box/j2cl/compose/J2clComposeSurfaceControllerTest.java:5031]: Fixed in 995a7f4 — the monster file is deleted; its tests are now split across 15 focused per-flow classes (each < 800 LOC, max 524) extending a shared J2clComposeControllerTestSupport base. All 182 tests compile and pass.

File: j2cl/src/test/java/org/waveprotocol/box/j2cl/compose/J2clComposeRichComponentTest.java [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+import org.waveprotocol.box.j2cl.search.J2clSidecarWriteSession;
+import org.waveprotocol.box.j2cl.transport.SidecarSubmitRequest;
+
+/**
+ * #1270: split out of the J2clComposeSurfaceControllerTest monster class — the
+ * rich-text component / annotation reply-submit path (J-UI-5, #1083). Uses the
+ * shared package doubles in {@link ComposeTestDoubles}.
+ */
```
[the remark this answers is not in the document]
Comment (vega113) [reply j2cl/src/test/java/org/waveprotocol/box/j2cl/compose/J2clComposeRichComponentTest.java:17]: Fixed in 995a7f4 — J2clComposeRichComponentTest now extends J2clComposeControllerTestSupport and its javadoc links to that real type instead of the non-existent ComposeTestDoubles.

File: j2cl/src/test/java/org/waveprotocol/box/j2cl/compose/J2clComposeRichComponentTest.java [hunk 4]
```diff
@@ -0,0 +1,289 @@
+package org.waveprotocol.box.j2cl.compose;
+
+import com.google.j2cl.junit.apt.J2clTestInput;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Test;
+import org.waveprotocol.box.j2cl.search.J2clSidecarWriteSession;
+import org.waveprotocol.box.j2cl.transport.SidecarSubmitRequest;
```
[the remark this answers is not in the document]
Comment (vega113) [reply j2cl/src/test/java/org/waveprotocol/box/j2cl/compose/J2clComposeRichComponentTest.java]: Fixed in 995a7f4 — per-file imports are recomputed; the unused Arrays and SidecarSubmitRequest imports are gone from J2clComposeRichComponentTest.
